### ISSUE TITLE
Syntax highlighting broken?

### ISSUE THREAD

**Issue by nickhudkins** [gandm/language-babel#251]
It appears that es7.decorators have stopped highlighting appropriately in recent updates.

**Comment by nickhudkins** [comment]
I can confirm that 2.37.1 behaves correctly. 2.38.0 does not.

**Comment by gandm** [comment]
I changed the scope names for decorators. What is the issue. Cab you beer more specific.
________________________________

**Comment by nickhudkins** [comment]
Ah, maybe it just is different, not broken 👍 . 

I guess I expected the decorator to be treated "as a whole" rather than the @ / methodName being separated. I didn't notice before that they were.

<img src="https://d3vv6lp55qjaqc.cloudfront.net/items/2n3k2D3X2t382L1n0z25/Screen%20Shot%202016-09-14%20at%209.41.48%20AM.png?X-CloudApp-Visitor-Id=35514&v=5010341" />

**Comment by gandm** [comment]
Yes I amended the decorators to support code like
```
@DecoratorClass
  .variable
  .actualDecorator
class Flo {}
```
I only know the real decorator when I've reached the class statement, and by that time I can't re-scope it using TextMate grammars. So I decided to just scope all lowercase dotted props including the decorator the same.

I may change this on the assumption that most people put the decorator, complete, on it's own line. In which case I'll code the last prop as the decorator as before.
